### ISSUE TITLE
Local grlc instance not working

### ISSUE THREAD

**Issue by marvinm2** [CLARIAH/grlc#219]
Hello,

I have been trying to get grlc working locally using the docker compose with my local folder of SPARQL queries (a git clone of https://github.com/marvinm2/AOPWikiQueries, locally in the path /home/marvin.martens/grlc/AOPWikiQueries). My SPARQL endpoint is a Virtuoso Docker image running on http://localhost:8890/sparql/. 

So, in the config.default.ini (from the cloned repo of grlc) I changed the local_sparql_dir and the sparql_endpoint to the correct ones. I left all other values unchanged. 

When running this docker-compose (docker-compose -f docker-compose.default.yml up), it starts, but is goes wrong when entering the local repo as described in https://github.com/CLARIAH/grlc/issues/90 (http://localhost:8001/api/local/local/) -- I couldn't find this information about using local folder anywhere else yet. When entering this link, I see an empty page, only containing logo and green bar, with "No API definition provided.". Did I miss something? I would like to get grlc working totally locally, with local sparql endpoint and query folder to not depend on github.

All help is appreciated

**Comment by c-martinez** [comment]
Hi @marvinm2 -- sorry for the delay in getting back to you. Just a quick question: you are using [this docker-compose file](https://github.com/CLARIAH/grlc/blob/master/docker-compose.default.yml)?

I'll look into your issue -- it might be that we need to fix a couple of things with the docker deploy instructions.

**Comment by marvinm2** [comment]
Correct, I cloned the grlc repository and I used that file. Are there things to edit there? I might have missed something in the file.

**Comment by c-martinez** [comment]
Ok, I need to check. I think we use this docker-compose file to spin up the grlc.io instance, which does not use the `/api/local/` option. It might be that this docker file just uses the image from docker-hub and does not load anything from your config file. At least that is my current theory.

**Comment by marvinm2** [comment]
When I run the docker-compose, I get a couple of warnings as well. A few of them have to do with a 'package.json' which cannot be found. 
```npm WARN deprecated connect@2.12.0: connect 2.x series is deprecated
npm WARN saveError ENOENT: no such file or directory, open '/home/marvin.martens/grlc/grlc/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/home/marvin.martens/grlc/grlc/package.json'
```

In the end of the docker-compose, it does actually read my config.ini file, pointing to my local SPARQL endpoint, but I don't see any notification about the local sparql directory.

**Comment by c-martinez** [comment]
Hi @marvinm2 -- I've tried figure out what is going on with docker-compose.default.yml file, but I think it might be broken (something to check with @albertmeronyo).

So I went a different way and just installed the latest version of grlc (from the dev branch), and started a local copy:
```
git clone https://github.com/marvinm2/AOPWikiQueries
<edit config.ini so it points to >
pip install "git+https://github.com/CLARIAH/grlc.git@dev"
grlc-server 
```
`config.ini` looks something like this:
```
...
[local]
local_sparql_dir = /AOPWikiQueries
...
```
and your api should be available on `http://localhost:8088/api/local/local/`.

Hope this helps!
